mne/cov.py

```diff
@@ -675,6 +675,38 @@ def _check_method_params(method, method_params, keep_sample_mean=True,
     return method, _method_params
 
 
+@verbose
+def compute_tfr_covariance(epochs_tfr, verbose=None, **kwargs):
+    """Compute covariance matrices across frequencies for TFR epochs.
+
+    Parameters
+    ----------
+    epochs_tfr : EpochsTFR
+        The time-frequency resolved epochs over which to compute the
+        covariance.
+    %(verbose)s
+    **kwargs : dict
+        Arguments to pass to :func:`mne.compute_covariance`.
+
+    Returns
+    -------
+    covs : list of Covariance
+        The covariance matrices for each frequency in the time-frequency
+        epochs.
+    """
+    from .time_frequency import EpochsTFR
+    _validate_type(epochs_tfr, EpochsTFR)
+    covs = list()
+    # move frequencies to front
+    for epochs_data in epochs_tfr.data.transpose([2, 0, 1, 3]):
+        epochs = EpochsArray(
+            epochs_data, epochs_tfr.info, events=epochs_tfr.events,
+            tmin=epochs_tfr.times[0] if epochs_tfr.times.size > 0 else None)
+        epochs.baseline = epochs_tfr.baseline
+        covs.append(compute_covariance(epochs, **kwargs))
+    return covs
```

Comment:
I would also return the frequencies  of each cov.

Reply:
not sure why this was marked "resolved" as it seems like the current code is still not returning the frequencies

Reply:
Good catch, I changed this for the `apply_*` functions but missed this one.